File: rb/features.rb
# AgentGateway SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AgentGatewayFeatures
  def self.make_feature(name)
    case name
    when "base"
      AgentGatewayBaseFeature.new
    when "ratelimit"
      AgentGatewayRatelimitFeature.new
    when "retry"
      AgentGatewayRetryFeature.new
    when "test"
      AgentGatewayTestFeature.new
    when "timeout"
      AgentGatewayTimeoutFeature.new
    else
      AgentGatewayBaseFeature.new
    end
  end
end

File: py/features.py
# AgentGateway SDK feature factory

from feature.base_feature import AgentGatewayBaseFeature
from feature.test_feature import AgentGatewayTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AgentGatewayBaseFeature(),
        "test": lambda: AgentGatewayTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
